src/io.c

```diff
@@ -630,8 +630,13 @@ mrb_io_sysread(mrb_state *mrb, mrb_value io)
   if (mrb_nil_p(buf)) {
     buf = mrb_str_new(mrb, NULL, maxlen);
   }
+
+  mrb_str_modify(mrb, RSTRING(buf));
+
   if (RSTRING_LEN(buf) != maxlen) {
     buf = mrb_str_resize(mrb, buf, maxlen);
+  } else {
+    mrb_str_modify(mrb, RSTRING(buf));
```

Comment:
Why do we call `mrb_str_modify` twice when RSTRING_LEN(buf) == maxlen?

Reply:
Oh, I forgot to remove the first one, of course, we don't need it.

I'll fix it.